.github/workflows/build.yml

```diff
@@ -0,0 +1,149 @@
+name: Build
+env:
+  GEOS_VERSION: "3.6.5"
+
+on:
+  push:
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+      - "packages/basemap_data/**"
+      - "packages/basemap_data_hires/**"
+  pull_request:
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+      - "packages/basemap_data/**"
+      - "packages/basemap_data_hires/**"
+  workflow_dispatch:
+
+jobs:
+  build_data:
+    name: Build data packages
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: [basemap_data, basemap_data_hires]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Install build dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build wheel
+
+      - name: Build package
+        run: |
+          cd packages/${{ matrix.package }}
+          python -m build --wheel --sdist
+
+      - uses: actions/upload-artifact@v4
+        with:
+          path: packages/${{ matrix.package }}/dist/*
+          name: dist-${{ matrix.package }}
+
+  build_basemap:
+    name: Build basemap wheels (${{ matrix.os }})
+    needs: [build_data]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            before_all: |
+              echo "Starting BEFORE_ALL script"
+              cd "{package}"
+              export GEOS_DIR="$(pwd)/extern"
+              echo "GEOS_DIR set to: $GEOS_DIR"
+              python -c "import sys; sys.path.append('utils'); from GeosLibrary import GeosLibrary; GeosLibrary('3.6.5', root='extern').build('extern', njobs=16)"
+            environment: >-
```

Comment:
This is just a comment for the future: The logs from building on GNU/Linux show a warning that `GEOS_DIR` was not defined or does not allow to find the GEOS library. My feeling here is that the `GEOS_DIR` variable needs to be defined again in this `environment` block and it is not enough to have it in the `before_all` block. The GNU/Linux build still succeeds because `CFLAGS` and `LDFLAGS` are hardcoded to the final correct values. A task *after* this PR would be to add `GEOS_DIR` here and see if the warning at build time disappears, and later remove `CFLAGS` and `LDFLAGS` and check that everything continues working.

Reply:
Yeah this stuff is giving me headaches. I was trying to debug when what is available. But I seem you have nicely fixed a few things I was running into too and failing to fix on a separate branch. Thanks!

Reply:
Resolving here because it was safe to remove `CFLAGS` and `LDFLAGS` after setting `GEOS_DIR`.